### DISCUSSION TITLE
C++ duckdb API to read a parquet file from an in-memory variable

### DISCUSSION THREAD

**Discussion by lkordos** [duckdb/duckdb#9142] [Q&A]
Hi,

Is it possible to use duckdb C++ API to import a parquet file to a duckdb database, if the file is stored in in-memory, binary buffer? All examples in documentation show import from an external file. I can't find any example how to do import from a memory location (a buffer holding a content of the parquet file).

Thanks.

**Comment by Mause** [comment]
This can be done with the C++ API, but you'll have to write an in memory filesystem implementation yourself, as DuckDB doesn't include one

**Comment by lkordos** [comment]
It seems to be too much work, especially if there should not be any file on a disk and the code should be portable across OSes. I was hoping it would be possible to pass to the duckdb API and address of the binary buffer with a parquet file content.
